python/cog/server/http.py

```diff
@@ -88,16 +89,42 @@ def create_app(
 
     predictor_ref = get_predictor_ref(config, mode)
 
+    try:
+        # TODO: avoid loading predictor code in this process
+        predictor = load_predictor_from_ref(predictor_ref)
+        InputType = get_input_type(predictor)
+        OutputType = get_output_type(predictor)
+    except Exception as ex:
+        app.state.health = Health.SETUP_FAILED
+        response = schema.PredictionResponse(input={},
+                                             output=None,
+                                             id=predictor_ref,
+                                             version=None,
+                                             created_at=datetime.now(),
+                                             started_at=datetime.now(),
+                                             completed_at=None,
+                                             logs="",
+                                             error=repr(ex),
+                                             status=schema.Status.FAILED,
+                                             metrics=None)
+        app.state.setup_result_payload = response
+
+        @app.get("/health-check")
+        async def healthcheck_startup_failed() -> Any:
```

Comment:
As mentioned in our chat, I think we still need a working `shutdown` endpoint when we fail this way.

Reply:
uploaded

Reply:
resolved by https://github.com/replicate/cog/pull/1433